### ISSUE TITLE
Init

### ISSUE THREAD

**Issue by felipe-dap** [felipe-dap/useTraversalThroughInputs#1]
@Suryakaran1234
So here we can talk about this hook.

**Comment by Suryakaran1234** [comment]
Okay, Thanks a lot btw! I also asked this question on stackoverflow, but nobody answered. So, I was feeling a little demoralized. I am glad to are helping me out on this.

**Comment by felipe-dap** [comment]
Cool.. Please send me the stackoverflow question url and I will post a answer there as well so anyone else can profit from this later.

**Comment by felipe-dap** [comment]
I made a scratch concept for you to follow along. The styling is very bad, but you will get the idea.
I dont know exactly if you will like the cells to be inputs for texting, or just <td> tags.
As for now it looks bad, but the arrow commands work nicely.
Also, openClose, I did not exactly understood your intentions with splice(1,10) so I left it behind.
If you explain me this I can help you out later on with open close.

**Comment by felipe-dap** [comment]
few things to take from this. subdividing your code in smaller chunks is a must.
life gets much easier with components.
importing components can be awful with ../../../components/TableCell stuff, so there is some cool tricks on customization that lets you point to root folder like this import {TableCell} from "~/components"
an important next step later you for your learnings is to use redux.
notice that we are passing props down to the components (in your example, current).
with redux you have this sort of library with your variables and any component can access it without you needing to pass them down as props. useContext does a similar job, but it think redux is more powerful. but that's another topic =)

**Comment by Suryakaran1234** [comment]
> Cool.. Please send me the stackoverflow question url and I will post a answer there as well so anyone else can profit from this later.

Here is the Link👇🏻 -
https://stackoverflow.com/questions/71840096/how-do-i-implement-a-function-to-navigate-through-my-react-table-using-arrow-key?noredirect=1#comment126967810_71840096

Sorry, I couldn't see this I was sick actually and took a day off from work.

**Comment by Suryakaran1234** [comment]
> I made a scratch concept for you to follow along. The styling is very bad, but you will get the idea. I dont know exactly if you will like the cells to be inputs for texting, or just tags. As for now it looks bad, but the arrow commands work nicely. Also, openClose, I did not exactly understood your intentions with splice(1,10) so I left it behind. If you explain me this I can help you out later on with open close.

Actually, My intention is to use those inputs for texting(in which users can insert values) and I used openClose to render inputs properly as they are in three lines(Open, Jodi and Close). In which, Open and Close are rendered beside jodi inputs. I hope it makes sense.
I have total 120 inputs (10 for Open in first column, 10 for Close in Last Column and Between them are 100 inputs for Jodi) in table cells.

**Comment by Suryakaran1234** [comment]
> few things to take from this. subdividing your code in smaller chunks is a must. life gets much easier with components. importing components can be awful with ../../../components/TableCell stuff, so there is some cool tricks on customization that lets you point to root folder like this import {TableCell} from "~/components" an important next step later you for your learnings is to use redux. notice that we are passing props down to the components (in your example, current). with redux you have this sort of library with your variables and any component can access it without you needing to pass them down as props. useContext does a similar job, but it think redux is more powerful. but that's another topic =)

Yeah, Actually this is my first project and I am trying very hard to use concepts that I have learned till now and learning everyday. I know about ContextApi which I think is a pretty good way to share state between components, but I am having a little hard time using Redux (maybe Because I don't know how it works properly).

**Comment by Suryakaran1234** [comment]
Hey, I have made this hook. Check this out and let me know if this is okay or not. Although it is working, I don't know if this is an appropriate way of dealing this.

**Here is the Hook Code👇🏻 -**

```
function useTraversalThroughInputs(initial) {
    const savedHandler = useRef();
    const UP_ARROW = 38;
    const DOWN_ARROW = 40;
    const LEFT_ARROW = 37;
    const RIGHT_ARROW = 39;
    const TOTAL_CELLS_BY_ROW = initial;

    const onKeyDown = (e) => {
        const index = e.target.parentElement.cellIndex;
        console.log(index);

        if (e.keyCode === LEFT_ARROW) {
            if (e.target.parentElement.previousElementSibling) {
                e.target.parentElement.previousElementSibling.children[1].focus();
            }
        }

        if (e.keyCode === RIGHT_ARROW) {
            if (e.target.parentElement.parentElement.nextElementSibling) {
                e.target.parentElement.nextElementSibling.children[1].focus();
            }
        }

        if (e.keyCode === DOWN_ARROW) {
            const nextRow = e.target.parentElement.parentElement.nextElementSibling;
            console.log(e.target.parentElement.parentElement.nextElementSibling.children[index % TOTAL_CELLS_BY_ROW].children[1]);
            if (nextRow) {
                e.target.parentElement.parentElement.nextElementSibling.children[index % TOTAL_CELLS_BY_ROW].children[1].focus();
            }
        }

        if (e.keyCode === UP_ARROW) {
            const previousRow = e.target.parentElement.parentElement.previousElementSibling;
            if (previousRow) {
                e.target.parentElement.parentElement.previousElementSibling.children[index % TOTAL_CELLS_BY_ROW].children[1].focus();
            }
        }
    };

    useEffect(() => {
        savedHandler.current = onKeyDown;
    }, []);

    const removeEvent = (element = window) => {
        // Make sure element supports addEventListener
        // On
        const isSupported = element && element.addEventListener;
        console.log(element, isSupported);
        if (!isSupported) return;
        // Create event listener that calls handler function stored in ref
        const eventListener = (event) => savedHandler.current(event);
        // Add event listener
        element.addEventListener('keydown', eventListener);
        // Remove event listener on cleanup
        return () => element.removeEventListener('keydown', eventListener);
    };

    return removeEvent;
}
```

**And That is How I am Using it in My Component👇🏻 -**  

  ```
const removeEvent = useTraversalThroughInputs(12); // 12 is the Number Of Columns of My Table in Table Body

useEffect(() => {
        removeEvent(document.querySelector('#table')); // '#table' is the id That I have given to my table body, as I have thought that I want to attach this event listener to it only and doesn't affect my rest web app.
    }, []);
```

**Comment by felipe-dap** [comment]
Sorry, @Suryakaran1234 I am really busy since last week.
When I have some time I will come back to have a look at your solution. =)
